CraftOrdersLib/data/scripts/player/map/mapcommands.lua

```diff
@@ -3,299 +3,343 @@
+		sellAmountTextBox.backgroundText = "Amount /* of goods to buy */"%_t
+		
+		local vsplit = UIVerticalSplitter(hsplit:partition(2), 10, 0, 0.7)
+		sellWindow:createLabel(vsplit.left, "Sell for at least X% of average price:"%_t, 14)
+		sellMarginCombo = sellWindow:createValueComboBox(vsplit.right, "")
+		
+		local vsplit = UIVerticalSplitter(hsplit.bottom, 10, 0, 0.5)
+		preferOwnStationsCheck = sellWindow:createCheckBox(vsplit.left, "Prefer Own Stations /* Checkbox caption for ship behavior */"%_t, "")
+		preferOwnStationsCheck.captionLeft = false
+		preferOwnStationsCheck.tooltip = "If checked, the ship will prefer your own stations for delivering the goods."%_t
+		
+		sellWindow:createButton(vsplit.right, "Sell /* Start sell order button caption */"%_t, "onSellWindowOKButtonPressed")
+		
+		
+		-- buying
+		buyWindow = GalaxyMap():createWindow(Rect(res * 0.5 - size * 0.5, res * 0.5 + size * 0.5))
+		buyWindow.caption = "Buy Goods /* Order Window Caption Galaxy Map */"%_t
+		
+		local hsplit = UIHorizontalMultiSplitter(Rect(buyWindow.size), 10, 10, 3)
+		local vsplit = UIVerticalMultiSplitter(hsplit.top, 10, 0, 1)
+		
+		buyCombo = buyWindow:createValueComboBox(vsplit.left, "")
+		
+		buyFilterTextBox = buyWindow:createTextBox(vsplit.right, "onBuyFilterTextChanged")
+		buyFilterTextBox.backgroundText = "Filter /* Filter Goods */"%_t
+		buyFilterTextBox.forbiddenCharacters = unmatchable
+		
+		local vsplit = UIVerticalSplitter(hsplit:partition(1), 10, 0, 0.7)
+		buyWindow:createLabel(vsplit.left, "Amount to have on ship:"%_t, 14)
+		
+		buyAmountTextBox = buyWindow:createTextBox(vsplit.right, "")
+		buyAmountTextBox.backgroundText = "Amount /* of goods to buy */"%_t
+		
+		local vsplit = UIVerticalSplitter(hsplit:partition(2), 10, 0, 0.7)
+		buyWindow:createLabel(vsplit.left, "Buy for at least X% of average price:"%_t, 14)
+		buyMarginCombo = buyWindow:createValueComboBox(vsplit.right, "")
+		
+		local vsplit = UIVerticalSplitter(hsplit.bottom, 10, 0, 0.5)
+		buyWindow:createButton(vsplit.right, "Buy /* Start buy order button caption */"%_t, "onBuyWindowOKButtonPressed")
+		
+		
+		-- both
+		for _, combo in pairs({buyMarginCombo, sellMarginCombo}) do
+			combo:addEntry(false, "Any"%_t)
+			for i = 50, 150, 5 do
+				combo:addEntry(i / 100, string.format("%i %%", i))
+			end
+		end
+		
+		
+		-- escort window
+		local escortSize = vec2(550, 50)
+		escortWindow = GalaxyMap():createWindow(Rect(res * 0.5 - escortSize * 0.5, res * 0.5 + escortSize * 0.5))
+		escortWindow.caption = "Escort Craft /* Order Window Caption Galaxy Map */"%_t
+		
+		local vsplit = UIVerticalSplitter(Rect(escortWindow.size), 10, 10, 0.6)
+		escortCombo = escortWindow:createValueComboBox(vsplit.left, "")
+		escortButton = escortWindow:createButton(vsplit.right, "Escort /* Start escort order button caption */"%_t, "onEscortWindowOKButtonPressed")
+		
+		
+		-- all windows
+		windows = {buyWindow, sellWindow, escortWindow}
```

Comment:
is the idea that this makes it possible to add more order interaction windows? (great idea by the way)

if so, what's the interaction that lets another modder add another window to this list?